api/db/changelog/db.changelog-97-cdr-archival-status.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+  <changeSet author="calbach" id="db.changelog-97-cdr-archival-status">
+    <addColumn tableName="cdr_version">
+      <column name="archival_status" type="tinyint" defaultValueNumeric="0">
```

Comment:
For the purposes of this PR - consistency. We don't use enum anywhere.

Generally though it's a good question. It would make debugging a lot simpler, but I haven't explored how this would interact with Hibernate, e.g. whether the corresponding Java enum values would be automatically validated. We'd also need to consider how hard/easy it is to rename values. In general I strongly prefer to keep code identifiers compartmentalized entirely from the storage representation, so I'd like that to be preserved. Here is some context on how we got to the current state: https://precisionmedicineinitiative.atlassian.net/browse/RW-872, though I didn't make any note of whether `enum` was considered and I don't recall now.

Reply:
Yeah, though the angle I would take would be "treat Swagger as code". And maybe regression tests for things we know we need to keep working.

See also this ticket on [less confusing](https://precisionmedicineinitiative.atlassian.net/browse/RW-3426) generated type names. Right now it's more of a gripe than a solution.

Reply:
I don't see how "treat swagger as code" has any bearing on this discussion.

Reply:
Mainly thinking of where it talks about reordering enum values. Right now we don't have as much standardization around maintaining the swagger files as we might want.

But yeah I can see how it's tricky if we're generating enums on the one hand in Swagger and on the other in the ORM from the DB. It'd be easy to get confused.

Reply:
In terms of the Swagger enums, standardization of the Swagger yaml itself was a secondary concern here. The more alarming reason to discard that solution was that Swagger codegen is a crap shoot - it's a completely underspecified open source best effort, I would not make any assumption of consistency in ordinal mapping or behaviors on enum changes across codegen versions, let alone languages.